matlab/.editorconfig

```diff
@@ -0,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This is a EditorConfig file: https://editorconfig.org/
+
+# See ../.editorconfig for inherited values
+
+[*.{c,cc,cpp,h,hh,hpp}]
+indent_size = 1
+indent_style = space
+
+[*.m]
+indent_size = 1
```

Comment:
1. Could we use the inherited `indent_size` of **2** for C++ files for consistency with the rest of the language bindings?
2. Could we set the `indent_size` to **4** for M files?

Reply:
Well, of course we can, but this doesn't match the current state of Matlab files, does it?

Reply:
To your point, I don't think we have a consistent indentation level which is programmatically enforced across the entire MATLAB code base at this time. I don't know that there is an officially supported linting tool available to enforce this via CI currently, but I am fully supportive of a solution like this if anyone has suggestions.

Regardless, the MATLAB editor uses a 4 space indent by default , so I thought it would make sense to use that in the `.editorconfig` file.

Reply:
Makes sense, thank you.